control-plane/pkg/reconciler/channel/channel.go

```diff
@@ -243,18 +240,15 @@ func (r *Reconciler) channelConfigMap() (*corev1.ConfigMap, error) {
 	return cm, nil
 }
 
-//nolint
-func (r *Reconciler) topicConfig(eventingKafkaConfig *commonconfig.EventingKafkaConfig) (*kafka.TopicConfig, error) {
-	// TODO: all hardcoded for now
+func (r *Reconciler) topicConfig(eventingKafkaConfig *commonconfig.EventingKafkaConfig, channel *messagingv1beta1.KafkaChannel) *kafka.TopicConfig {
 	return &kafka.TopicConfig{
 		TopicDetail: sarama.TopicDetail{
-			NumPartitions:     1,
-			ReplicationFactor: 1,
-			ReplicaAssignment: nil,
-			ConfigEntries:     nil,
+			NumPartitions:     channel.Spec.NumPartitions,
+			ReplicationFactor: channel.Spec.ReplicationFactor,
+			// TODO: retention period from the spec is not used
```

Comment:
This is to be done later. It will probably cause changes in bunch of places.

Reply:
> It will probably cause changes in bunch of places.

Not sure why is this the case, should we just set it in `ConfigEntries`? Like here

https://github.com/knative-sandbox/eventing-kafka/blob/acf06a29bf4450a140bc4164a1882a5bfb4a99c6/pkg/channel/consolidated/reconciler/controller/kafkachannel.go#L536-L549

Anyway,
/lgtm

Reply:
Hm ok, I will try that in a separate PR

Reply:
I thought I needed to modify `TopicDetail` struct, but it is not "our" struct 🤦 

Thanks for the heads up!

Separate PR here: https://github.com/knative-sandbox/eventing-kafka-broker/pull/1212